src/common/wallet-state.ts

```diff
@@ -319,15 +321,34 @@ function useWalletState(initialState?: Partial<WalletStateParams>): WalletData {
   const getCurrentPrivateKey = (password: string): string =>
     decryptCurrentAccount(password).privateKey
 
+  const fetchBalance = async (address: string): Promise<Option<Wei>> =>
+    web3.eth
+      .getBalance(address, 'latest')
+      .then(asWei)
+      .then(
+        (balance) => option.some(balance),
+        (err) =>
+          pipe(
+            err.data as unknown,
+            CustomErrors.decode,
+            option.fromEither,
+            option.filter((customErrors) => customErrors.some((error) => error.code === 100)),
+            option.fold(
+              () => Promise.reject(err),
+              (_) => Promise.resolve(option.none),
```

Comment:
Does this mean that if any of the errors has the code `100`, the promise gets resolved?

Reply:
yes, with a none, because 100 is code for missing data